src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -458,6 +461,49 @@ async function getAccessCodeGatedPlatforms(
   return validateResult.platforms
 }
 
+async function getTokenGatedPlatforms(channel_id: number): Promise<PlatformsMetaInterface> {
+  if (!window.ethereum) throw 'Window.ethereum provider not found'
+  const provider = new ethers.BrowserProvider(window.ethereum)
+  const signer = await provider.getSigner()
```

Comment:
**Optional** but maybe we can have this as a helper function

```typescript
async function getSigner(): Promise<ethers.Signer> {
  if (!window.ethereum) {
    throw new Error('Ethereum provider not found');
  }
  const provider = new ethers.BrowserProvider(window.ethereum);
  return provider.getSigner();
}
```

Reply:
Great idea, I broke this logic out into getSigner!